carcara/src/ast/pool/mod.rs

```diff
@@ -184,6 +237,54 @@ impl PrimitivePool {
                 result.push(self.compute_sort(body));
                 Sort::Function(result)
             }
+            Term::IndexedOp { op, op_args, args } => {
+                let sort = match op {
+                    IndexedOperator::BvExtract => {
+                        /*
+                        ! extract has restrictions regarding their op_args
+                        ((_ extract i j) (_ BitVec m) (_ BitVec n))
+
+                        where
+                        - i, j, m, n are numerals
+                        - m > i ≥ j ≥ 0,
+                        - n = i - j + 1
+                         */
+                        let Constant::Integer(i) = op_args[0].clone() else {
+                            todo!()
+                        };
+                        let Constant::Integer(j) = op_args[1].clone() else {
+                            todo!()
+                        };
+
+                        let Sort::BitVec(m) =
+                            self.compute_sort(&args[0]).as_sort().unwrap().clone()
+                        else {
+                            unreachable!()
+                        };
+                        if !(m > i && i >= j && j >= Integer::ZERO) {
+                            panic!("Wrong arguments for extract") // todo create error for this
+                        }
+                        Sort::BitVec(i - j + Integer::ONE)
+                    }
+                    IndexedOperator::ZeroExtend | IndexedOperator::SignExtend => {
+                        let Constant::Integer(extension_width) = op_args[0].clone() else {
```

Comment:
By the way, it might be nice to add some helper methods like `Constant::as_integer` (that returns an `Option<Integer>`) so you can just call `op_args[0].as_integer().unwrap()`. Something similar to the `Term::as_{var,sort,etc}` methods.